### PR TITLE
 Let CupertinoTabScaffold's tab be programmatically selectable

### LINKED ISSUE #15969: CupertinoTabScaffold current index
CupertinoTabScaffold should support the current index using CupertinoTabBar value. 

How can I change current index on CupertinoTabScaffold?

### PR DESCRIPTION
**Pull request by xster**
Fixes #15969

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/tab_scaffold.dart [hunk 1]
```diff
@@ -121,14 +128,30 @@ class CupertinoTabScaffold extends StatefulWidget {
...
+  void initState() {
+    super.initState();
+    _currentPage = widget.tabBar.currentIndex;
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTabScaffold oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.tabBar.currentIndex != oldWidget.tabBar.currentIndex) {
+      setState(() => _currentPage = widget.tabBar.currentIndex);
```
Comment (HansMuller) [inline packages/flutter/lib/src/cupertino/tab_scaffold.dart]: You don't need to call `setState` here.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/cupertino/tab_scaffold.dart (hunk 1 above)
Comment (xster) [reply packages/flutter/lib/src/cupertino/tab_scaffold.dart]: ya, totally. Thanks!

(we should add a debug time assert or a linter :))

**Comment by Andy1984** [comment]
Is this already merged? I still cannot find currentTabIndex.

**Pull request merged**